server/src/main/java/com/defold/extender/services/GradleService.java

```diff
@@ -139,21 +145,21 @@ private void createGradlePropertiesFile(File gradlePropertiesFile, Boolean useJe
         Files.write(gradlePropertiesFile.toPath(), contents.getBytes());
     }
 
-    private void createLocalPropertiesFile(File localPropertiesFile) throws IOException {
+    private void createLocalPropertiesFile(File localPropertiesFile, Map<String, Object> jobEnvContext) throws IOException {
         HashMap<String, Object> envContext = new HashMap<>();
-        envContext.put("android-sdk-root", ANDROID_SDK_ROOT);
+        envContext.put("android-sdk-root", jobEnvContext.get("env.ANDROID_SDK_ROOT").toString());
         String contents = templateExecutor.execute(localPropertiesTemplateContents, envContext);
         Files.write(localPropertiesFile.toPath(), contents.getBytes());
     }
 
-    private void createBuildGradleFile(File mainGradleFile, List<File> gradleFiles) throws IOException {
+    private void createBuildGradleFile(File mainGradleFile, List<File> gradleFiles, Map<String, Object> jobEnvContext) throws IOException {
         List<String> values = new ArrayList<>();
         for (File file : gradleFiles) {
             values.add(file.getAbsolutePath());
         }
         HashMap<String, Object> envContext = new HashMap<>();
         envContext.put("gradle-files", values);
-        envContext.put("compile-sdk-version", ANDROID_SDK_VERSION);
+        envContext.put("compile-sdk-version", jobEnvContext.get("env.ANDROID_SDK_VERSION").toString());
```

Comment:
Why the need for toString()?
The context contains strings, or lists of strings.
And, you're passing from one Object to another Object, no need to get the string explicitly?